Replace Travis tests with GitHub Actions

AyanSinhaMahapatra opened the pull request:
Remove travis and add github actions for
pytest and documentation tests.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

pombredanne commented on .github/workflows/ci-pytest.yml:
```diff
@@ -0,0 +1,32 @@
+        run: source tmp/bin/activate
+
+      - name: Run tests
+        run: tmp/bin/pytest -vvs
```
Since you call a path here there is no need to activate above.

AyanSinhaMahapatra replied:
Yes, changes made.

pombredanne reviewed (approved):
LGTM!
I wonder though if using Azure would make more sense?
Are you using the skeletton here?

AyanSinhaMahapatra commented:
@pombredanne yes this has skeleton #26, and I modified it to only have run the tests on linux #31, as you asked of.
So yes azure test files are there [](https://github.com/nexB/scancode-results-analyzer/blob/master/etc/ci/azure-linux.yml)
I think you'd have to enable the builds for this repository?  

Also, the GitHub actions work fine, I tested on my fork. 

- The [pytest fails with exit code 5](https://github.com/AyanSinhaMahapatra/scancode-results-analyzer/runs/1888387749?check_suite_focus=true), because no tests are collected yet. #32 Has the tests.
- The [documentation test fails](https://github.com/AyanSinhaMahapatra/scancode-results-analyzer/runs/1888387795?check_suite_focus=true) because of an error in the requirements file, fixed [here](https://github.com/nexB/scancode-results-analyzer/pull/33/commits/1bc2d4176b695d8051ae8736e9afb081e47a297d) in #33.

So, after this PR is pushed the PRs #32 and #33 would pass these tests.

pombredanne reviewed (approved):
LGTM, merging... though as mentioned, it may help to use Azure pipelines too.
